lib/python/picongpu/pypicongpu/walltime.py

```diff
@@ -0,0 +1,30 @@
+"""
+This file is part of the PIConGPU.
+Copyright 2024-2024 PIConGPU contributors
+Authors: Brian Edward Marre, Richard Pausch
+License: GPLv3+
+"""
+
+import pydantic
+import datetime
+
+from .rendering import RenderedObject
+
+
+class Walltime(RenderedObject, pydantic.BaseModel):
+    walltime: datetime.timedelta
+    """time after which the cluster scheduler will stop the simulation"""
+
+    def check(self) -> None:
+        if self.walltime.total_seconds() <= 0.0:
+            raise ValueError("walltime must be > 0.")
+
+    def _get_serialized(self) -> dict:
+        return {
+            "walltime": "{:d}:{:02d}:{:02d}".format(
+                int(self.walltime.total_seconds() // 3600),  # hours
+                int(self.walltime.total_seconds() % 3600 // 60),  # minutes
+                int(self.walltime.total_seconds() % 3600),  # seconds
+            )
```

Comment:
How to do this according to [Python docs](https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds):
```suggestion
            "walltime": "{:d}:{:02d}:{:02d}".format(
                (r := divmod(self.walltime, HOUR))[0],  # hours
                (r := divmod(r[1], MINUTE))[0],  # minutes
                (r[1] // SECOND),  # seconds
            )
```
with global constants
```python
HOUR = timedelta(hours=1)
MINUTE = timedelta(minutes=1)
SECOND = timedelta(seconds=1)
```
PS: Your seconds calculation has a bug, so even if you don't fancy that other method, please fix:
```suggestion
            "walltime": "{:d}:{:02d}:{:02d}".format(
                int(self.walltime.total_seconds() // 3600),  # hours
                int(self.walltime.total_seconds() % 3600 // 60),  # minutes
                int(self.walltime.total_seconds() % 60),  # seconds
            )
```

Reply:
Good Catch!